marketstack/src/adapter.ts

```diff
@@ -0,0 +1,41 @@
+import { Requester, Validator, AdapterError } from '@chainlink/external-adapter'
+import { Config, ExecuteWithConfig, ExecuteFactory } from '@chainlink/types'
+import { makeConfig, DEFAULT_ENDPOINT } from './config'
+import { eod } from './endpoint'
+
+const inputParams = {
+  endpoint: false,
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, config) => {
+  const validator = new Validator(request, inputParams)
+  if (validator.error) throw validator.error
+
+  Requester.logConfig(config)
+
+  const jobRunID = validator.validated.id
+  const endpoint = validator.validated.data.endpoint || DEFAULT_ENDPOINT
+
+  switch (endpoint) {
+    case eod.Name: {
+      const data = await eod.execute(config, request)
+
+      return Requester.success(jobRunID, {
```

Comment:
This will quickly grow to be hard to read if say 10 other endpoints were added.

I think the best style would be how Jonas modified the `example` adapter as seen here: 
https://github.com/smartcontractkit/external-adapters-js/blob/develop/example/src/endpoint/example.ts#L32
Each endpoint handles its own return, which keeps `adapter.ts` a lot cleaner.